### ISSUE TITLE
GZIP:invalid compressed data--format violated

### ISSUE THREAD

**Issue by btrspg** [OpenGene/fastp#180]
Hi there,
     I found the output of fastp was something wrong
     `gzip: xxx.clean.2.fq.gz: invalid compressed data--format violated` when I used `zcat xxx.clean.2.fq.gz`; 
     But there was no any error log during the fastp running time, So I wonder if the fastp can provide some check if the output Is really what it supposed to output.

Thanks!

**Comment by sfchen** [comment]
But you can gunzip it, right?

I will check it

**Comment by btrspg** [comment]
@sfchen Yes, but the reads number is different from the reads_number_after_filter in json file. And read1 number is not same as read2 number.It seems like the fa.gz did not complete!

**Comment by hreypar** [comment]
Same here, FastQC and other downstream programs refuse to accept fastp .gz output. "invalid gzip format"

**Comment by sfchen** [comment]
What's your OS and fastp version? Is it in cloud or in local server?

**Comment by hreypar** [comment]
> What's your OS and fastp version? Is it in cloud or in local server?

"Ubuntu 18.04.2 LTS (Bionic Beaver)"
fastp 0.20.0 

it is a local machine. 

Thanks!

**Comment by sfchen** [comment]
Can you confirm that the fastp process has finished correctly?

**Comment by hreypar** [comment]
> Can you confirm that the fastp process has finished correctly?

I ended up printing to stdout and piping to gzip, so I didn't use fastp's internal gzip, but fastp's processing of the reads worked out just fine, thanks!

fastp -i  $prereq \
		--stdout \
		--json results/$stem.json \
		--html results/$stem.html \
	| gzip \
		-f \
		--best \
	> $target \

**Comment by btrspg** [comment]
@sfchen fastp process finished correctly, with not any error in the log file. 
Here is the log file
```
Read1 before filtering:
total reads: 33843222
total bases: 5076483300
Q20 bases: 4932327219(97.1603%)
Q30 bases: 4585968494(90.3375%)

Read2 before filtering:
total reads: 33843222
total bases: 5076483300
Q20 bases: 4890775279(96.3418%)
Q30 bases: 4505376147(88.7499%)

Read1 after filtering:
total reads: 33571200
total bases: 5027181154
Q20 bases: 4895952444(97.3896%)
Q30 bases: 4554343615(90.5944%)

Read2 aftering filtering:
total reads: 33571200
total bases: 5027181154
Q20 bases: 4858594974(96.6465%)
Q30 bases: 4479078663(89.0972%)

Filtering result:
reads passed filter: 67142400
reads failed due to low quality: 154266
reads failed due to too many N: 389778
reads failed due to too short: 0
reads with adapter trimmed: 1207960
bases trimmed due to adapters: 17008918
reads corrected by overlap analysis: 7810769
bases corrected by overlap analysis: 12483875

Duplication rate: 2.91815%

Insert size peak (evaluated by paired-end reads): 215

JSON report: xxx.fastp.json
HTML report: xxx.fastp.html

fastp -c -i xxx_1.fq.gz -I xxx_2.fq.gz -o xxx.clean.1.fq.gz -O xxx.2.fq.gz --html xxx.fastp.html --json xxx.fastp.json
fastp v0.20.0, time used: 374 seconds
```

I think if the fastp could comfirm the output file before it really finish.
Thanks.

**Comment by ellisellen** [comment]
I have the exact same issue for some fastq files

**Comment by btrspg** [comment]
Hi @ellisellen, as the `fastp` hasn't fixed this issue, I used `gzip -t` to test if output `fq.gz` files were completed. Hope this would help you.

```
fastp xxxxxxx && gzip -t read1_clean_fq.gz && gzip -t read2_clean_fq.gz
```

Even this only check the completion of the gzip process, it helped me from going to the downstream analysis without any error raise.
